From c305f6ede17e993e5e662dc456e690cf1b1380d8 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 25 Nov 2018 18:59:00 +0200
Subject: [PATCH] version: provide version for the upper s/w layers

Provide version information for the upper s/w layers.
For this we place the version string in a global variable and write its
offset in SRAM to the specially reserved scratch pad 28 register.
Reuse the variable for preboot version printing.

Change-Id: I0880322368e9241b036587448fb6e6191f1773d2
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot.c b/preboot.c
index 550927c..3dc0f5e 100644
--- a/preboot.c
+++ b/preboot.c
@@ -53,6 +53,9 @@
 #define SKIP_PCIE_INIT BIT(16)
 #endif
 
+#define PREBOOT_VER_STR	"Preboot version " PREBOOT_VER \
+	" (" PREBOOT_DATE " - " PREBOOT_TIME ")"
+
 static void hl_preboot_puts_raw(const char* str)
 {
 	int i;
@@ -456,17 +459,19 @@
 	hl_preboot_puts("PCIe link is UP!\n");
 }
 
+static const char *preboot_ver_str = PREBOOT_VER_STR;
+
 #ifndef TARGET_IS_TESTER
 static void hl_preboot_version(void)
 {
+	u64 offset = (u64) preboot_ver_str;
+
 	hl_preboot_puts("\n");
-	hl_preboot_puts_raw("Preboot ");
-	hl_preboot_puts_raw(PREBOOT_VER);
-	hl_preboot_puts_raw(" (");
-	hl_preboot_puts_raw(PREBOOT_DATE);
-	hl_preboot_puts_raw(" - ");
-	hl_preboot_puts_raw(PREBOOT_TIME);
-	hl_preboot_puts(")\n");
+	hl_preboot_puts_raw(preboot_ver_str);
+	hl_preboot_puts("\n");
+
+	writel((u32)(offset & ~CONFIG_SYS_SDRAM_BASE),
+			mmPSOC_GLOBAL_CONF_SCRATCHPAD_28);
 }
 #else	/* TARGET_IS_TESTER */
 static inline void hl_preboot_version(void) {}
